net_transport.go

```diff
@@ -872,7 +896,9 @@ func (n *netPipeline) AppendEntries(args *AppendEntriesRequest, resp *AppendEntr
 
 	// Add a send timeout
 	if timeout := n.trans.timeout; timeout > 0 {
-		n.conn.conn.SetWriteDeadline(time.Now().Add(timeout))
+		if err := n.conn.conn.SetWriteDeadline(time.Now().Add(timeout)); err != nil {
+			return nil, err
+		}
```

Comment:
How does this change behavior in the caller? We're changing the error handling in _many_ places in this PR.  We're doing this for the sake of linting without much in the way of consideration for how it changes the expected behavior. "It silences linters" isn't a good enough reason to make this change without that consideration.

(This comment applies to a lot of these.)